receiver/prometheusreceiver/config.go

```diff
@@ -95,6 +95,10 @@ func checkTLSConfig(tlsConfig commonconfig.TLSConfig) error {
 // Validate checks the receiver configuration is valid.
 func (cfg *Config) Validate() error {
 	promConfig := cfg.PrometheusConfig
+	if (promConfig == nil || len(promConfig.ScrapeConfigs) == 0) && cfg.TargetAllocator == nil {
+		return errors.New("no Prometheus scrape_configs or target_allocator set")
+	}
+
 	if promConfig != nil {
```

Comment:
I would suggest moving this to the `validatePromConfig` simplifies indentation.

Reply:
That's where it was and you told me to move it.

Reply:
I told to move the following code in the config.Validate

```
	if (promConfig == nil || len(promConfig.ScrapeConfigs) == 0) && cfg.TargetAllocator == nil {
		return errors.New("no Prometheus scrape_configs or target_allocator set")
	}
```

And in this comment I asked to move in the `validatePromConfig`
```
	if promConfig == nil {
		return nil
	}
```